Source/CarthageKit/Archive.swift

```diff
@@ -36,9 +36,41 @@ public func unzip(archive fileURL: URL, to destinationDirectoryURL: URL) -> Sign
 		.then(SignalProducer<(), CarthageError>.empty)
 }
 
+/// Untars the gzipped archive at the given file URL, extracting into the given
+/// directory URL (which must already exist).
+public func untargz(archive fileURL: URL, to destinationDirectoryURL: URL) -> SignalProducer<(), CarthageError> {
+	precondition(fileURL.isFileURL)
+	precondition(destinationDirectoryURL.isFileURL)
+
+	let task = Task("/usr/bin/env", arguments: [ "tar", "-xzf", fileURL.carthage_path, "-C", destinationDirectoryURL.carthage_path ])
+	return task.launch()
+		.mapError(CarthageError.taskError)
+		.then(SignalProducer<(), CarthageError>.empty)
+}
+
+
 /// Unzips the archive at the given file URL into a temporary directory, then
 /// sends the file URL to that directory.
 public func unzip(archive fileURL: URL) -> SignalProducer<URL, CarthageError> {
+	return createTemporaryDirectory()
+		.flatMap(.merge) { directoryURL in
+			return unzip(archive: fileURL, to: directoryURL)
+				.then(SignalProducer<URL, CarthageError>(value: directoryURL))
+		}
+}
+
+
+/// Untars the gzipped archive at the given file URL into a temporary directory, 
+/// then sends the file URL to that directory.
+public func untargz(archive fileURL: URL) -> SignalProducer<URL, CarthageError> {
```

Comment:
I think we can make `untargz` and `unzip` private if we add a public `unarchive`.